### PR TITLE
migrate 9pfs to use TextEncoder to fix 3 byte utf bug

### PR DESCRIPTION
**Pull request by ProgrammerIn-wonderland**
3 byte utf8 characters such as `【` in 9pfs filenames are currently broken due to a bugged UTF implementation in /lib/utf8.js. This PR aims to get rid of most of the code in utf8.js and replace it with the modern TextEncoder and TextDecoder equivalents

### CODE REVIEW

File: lib/utf8.js [hunk 1]
```diff
@@ -4,60 +4,9 @@
...
-        return -1;
-    };
-}
-
-function UnicodeToUTF8Stream(key)
-{
-        if(key < 0x80)  return [key];
-        if(key < 0x800) return [0xC0|((key>>6)&0x1F), 0x80|(key&0x3F)];
+function utf8_len(s) {
+    return texten.encode(s).length;
```
Comment (copy) [inline lib/utf8.js:11]: I'd suggest to inline this function and delete this file.

Comment (ProgrammerIn-wonderland) [reply lib/utf8.js:11]: :+1:, what's a good new home for it, 9p.js?

Comment (ProgrammerIn-wonderland) [reply lib/utf8.js:11]: Got it

**Comment by copy** [comment]
Thanks!

**Pull request merged**